contracts/strategies/AvaiStrategy.sol

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+
+import "../interfaces/IAvaiPodLeader.sol";
+import "../interfaces/IERC20.sol";
+import "../interfaces/IPair.sol";
+import "../lib/DexLibrary.sol";
+import "./MasterChefStrategyV1.sol";
+
+contract AvaiStrategy is MasterChefStrategyV1 {
+    using SafeMath for uint256;
+
+    IAvaiPodLeader public podLeader;
+    address public nativeRewardToken;
+    address public swapPairRewardToken;
+    uint256 public depositFeeBips;
+
+    constructor(
+        string memory _name,
+        address _depositToken,
+        address _rewardToken,
+        address _nativeRewardToken,
+        address _swapPairToken0, // swap rewardToken to token0
+        address _swapPairToken1, // swap rewardToken to token1
+        address _swapPairRewardToken, // swap nativeRewardToken to rewardToken
+        address _stakingRewards,
+        uint256 _pid,
+        address _timelock,
+        uint256 _depositFeeBips,
+        uint256 _minTokensToReinvest,
+        uint256 _adminFeeBips,
+        uint256 _devFeeBips,
+        uint256 _reinvestRewardBips
+    )
+        Ownable()
+        MasterChefStrategyV1(
+            _name,
+            _depositToken,
+            _rewardToken,
+            _swapPairToken0,
+            _swapPairToken1,
+            _stakingRewards,
+            _timelock,
+            _pid,
+            _minTokensToReinvest,
+            _adminFeeBips,
+            _devFeeBips,
+            _reinvestRewardBips
+        )
+    {
+        podLeader = IAvaiPodLeader(_stakingRewards);
+        depositFeeBips = _depositFeeBips;
+        nativeRewardToken = _nativeRewardToken;
+        if (nativeRewardToken != address(rewardToken)) {
+            swapPairRewardToken = _swapPairRewardToken;
+            require(
+                isPairEquals(
+                    IPair(swapPairRewardToken),
+                    nativeRewardToken,
+                    address(rewardToken)
+                ) ||
+                    isPairEquals(
+                        IPair(swapPairRewardToken),
+                        address(rewardToken),
+                        nativeRewardToken
+                    ),
+                "Swap pair does not match reward token and native reward token"
+            );
+        }
+    }
+
+    function isPairEquals(
```

Comment:
DexLibrary already has a function that does that and it does even better, it check if right and left are not mismatching and it doesn't mind position(left or right), it takes it into tokenA tokenB perspective, so it's a good idea to use that